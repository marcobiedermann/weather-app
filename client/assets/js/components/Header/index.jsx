import PropTypes from 'prop-types';
import React from 'react';

const Header = props => {
  const { children, ...otherProps } = props;

  return <header {...otherProps}>{children}</header>;
};

Header.propTypes = {
  children: PropTypes.node,
};

Header.defaultProps = {
  children: null,
};

export default Header;
