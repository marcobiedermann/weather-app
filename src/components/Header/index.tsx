import { ReactNode } from 'react';
import styles from './style.module.css';

interface HeaderProps {
  children: ReactNode;
}

function Header(props: HeaderProps): JSX.Element {
  return <header className={styles.header} {...props} />;
}

export type { HeaderProps };
export default Header;
