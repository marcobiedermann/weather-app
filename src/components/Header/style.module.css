.header {
  border-bottom: 1px solid #eef2f3;
  padding: 3em;
}
