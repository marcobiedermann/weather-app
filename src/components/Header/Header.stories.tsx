import type { Meta, StoryObj } from '@storybook/react';
import Header from '.';

const meta: Meta<typeof Header> = {
  args: {
    children: 'Header',
  },
  component: Header,
};

type Story = StoryObj<typeof Header>;

const Default: Story = {};

export { Default };
export default meta;
