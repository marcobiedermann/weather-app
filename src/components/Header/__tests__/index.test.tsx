import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import Header from '..';

describe('components/Header', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(<Header>Content</Header>);

    expect(asFragment()).toMatchSnapshot();
  });
});
