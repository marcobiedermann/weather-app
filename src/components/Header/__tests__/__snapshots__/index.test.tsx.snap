// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/Header > renders correctly 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    Header
  </header>
</DocumentFragment>
`;
